### DISCUSSION TITLE
Url Encoding Micropython

### DISCUSSION THREAD

**Discussion by James-McCracken** [micropython/micropython#9763] [General]
Programming a Raspberry Pi Pico W and I need to url encode the parameters. It does not seem to have access to the urllib module and urequests.request does not allow for the params parameter. does anyone have any suggestions?

**Comment by scruss** [comment]
It's not available in the Pico W port. The [urllib.parse](https://github.com/micropython/micropython-lib/tree/f3cfc52ab076fc913dc6e266fb7370b418c8f542/unix-ffi/urllib.parse) library only works on the Unix port.

**Comment by xuancong84** [comment]
`urlencode()` is very useful. I also need it on ESP8266 and ESP32. Please add it. Thanks!
